src/devices/multipleAnalogSensorsRosPublishers/GenericSensorRosPublisher.h

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (C) 2006-2019 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+
+#ifndef YARP_DEV_GENERICSENSORSROSPUBLISHER_H
+#define YARP_DEV_GENERICSENSORSROSPUBLISHER_H
+
+#include <yarp/os/PeriodicThread.h>
+#include <yarp/os/Publisher.h>
+#include <yarp/os/Node.h>
+#include <yarp/dev/DeviceDriver.h>
+#include <yarp/dev/IMultipleWrapper.h>
+#include <yarp/dev/MultipleAnalogSensorsInterfaces.h>
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+/**
+ * @ingroup dev_impl_wrapper
+ *
+ * \brief This abstract template needs to be specialized in a ROS Publisher, for a specific ROS mesagge/sensor type.
+ *
+ * | YARP device name |
+ * |:-----------------:|
+ * | `GenericSensorRosPublisher` |
+ *
+ * The parameters accepted by this device are:
+ * | Parameter name | SubParameter   | Type    | Units          | Default Value | Required                    | Description                                                       | Notes |
+ * |:--------------:|:--------------:|:-------:|:--------------:|:-------------:|:--------------------------: |:-----------------------------------------------------------------:|:-----:|
+ * | name           |      -         | string  | -              |   -           | Yes                         | Prefix of the port opened by this device                          | MUST start with a '/' character |
+ * | period         |      -         | int     | ms             |   -           | Yes                          | Refresh period of the broadcasted values in ms                    |  |
```

Comment:
I may be wrong, but at some point we did not agree to start switching devices and modules to express periods in seconds?

Reply:
I remember the same. However, this piece of code is copied from `MultipleAnalogSensorsServer` (not written by me), so I'm not sure anymore. What do we do? Should we change it, too?

Reply:
> but at some point we did not agree to start switching devices and modules to express periods in seconds?

Yes, the new devices should use seconds.

Reply:
> 
> 
> > but at some point we did not agree to start switching devices and modules to express periods in seconds?
> 
> Yes, the new devices should use seconds.

Using the same usual `period` name or a different parameter name? There is a reference to the relevant guideline/discussion?

Reply:
`period` is fine, unfortunately I cannot find the discussion, maybe we did it in a yarp council without writing it down?
In this case we should write a sort of announcement.

Reply:
Ack, even if mixing `period` parameters of different device with different units risk to be error-prone.

Reply:
So I'm going to change this to seconds.
I will not change `MultipleAnalogSensorsServer` and other similar devices since they might be already in use by people. Please proceed with the announcement...